channels/drdynvc/client/drdynvc_main.c

```diff
@@ -336,83 +343,128 @@ static UINT dvcman_load_addin(drdynvcPlugin* drdynvc, IWTSVirtualChannelManager*
 	return ERROR_INVALID_FUNCTION;
 }
 
-static DVCMAN_CHANNEL* dvcman_channel_new(drdynvcPlugin* drdynvc,
-                                          IWTSVirtualChannelManager* pChannelMgr, UINT32 ChannelId,
-                                          const char* ChannelName)
+static void dvcman_channel_free(DVCMAN_CHANNEL* channel)
 {
-	DVCMAN_CHANNEL* channel;
-
-	if (dvcman_find_channel_by_id(pChannelMgr, ChannelId))
-	{
-		WLog_Print(drdynvc->log, WLOG_ERROR,
-		           "Protocol error: Duplicated ChannelId %" PRIu32 " (%s)!", ChannelId,
-		           ChannelName);
-		return NULL;
-	}
+	if (!channel)
```

Comment:
Think this should be an `assert`

Reply:
Hum yes correct, that's the rest of the previous implementation, `dvcman_channel_free` was used as dtor of the `channelById` Hashtable, and it happens that the dtor is called with NULL. Will change that and see if that's the only case.

Reply:
I did the changes.